extensions/wikia/TemplateDraft/ApprovedraftAction.php

```diff
@@ -96,10 +98,12 @@ private function approveDraft( Title $draftTitle ) {
 		$draftPage = WikiPage::newFromID( $draftTitle->getArticleID() );
 		$draftPage->doDeleteArticle( wfMessage( 'templatedraft-draft-removal-summary' )->inContentLanguage()->plain() );
 
-		// Update Insights list
-		$model = InsightsHelper::getInsightModel( InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE );
-		if ( $model instanceof InsightsQuerypageModel ) {
-			$model->updateInsightsCache( $parentTitle->getArticleID() );
+		// Update Insights list if enabled
+		if ( $wgEnableInsightsInfoboxes ) {
+			$model = InsightsHelper::getInsightModel( InsightsUnconvertedInfoboxesModel::INSIGHT_TYPE );
+			if ( $model instanceof InsightsQuerypageModel ) {
```

Comment:
Can we have `getInsightModel` always return an object that satisfies an interface that includes `updateInsightsCache`? 

Also it looks like the introduction of `$wgEnableInsightsInfoboxes` is leaking implementation details from `InsightsHelper` into this class. Could that behavior be encapsulated in `updateInsightsCache`?